packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,448 @@
+import route53 = require('@aws-cdk/aws-route53');
+import cdk = require('@aws-cdk/cdk');
+import { AliasTargetInstance } from './alias-target-instance';
+import { CnameInstance, CnameInstanceBaseProps  } from './cname-instance';
+import { IInstance } from './instance';
+import { IpInstance, IpInstanceBaseProps } from './ip-instance';
+import { INamespace, NamespaceType } from './namespace';
+import { NonIpInstance, NonIpInstanceBaseProps } from './non-ip-instance';
+import { CfnService } from './servicediscovery.generated';
+
+export interface IService extends cdk.IConstruct {
+  /**
+   * A name for the Cloudmap Service.
+   */
+  readonly serviceName: string;
+
+  /**
+   *  The namespace for the Cloudmap Service.
+   */
+  readonly namespace: INamespace;
+
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  readonly serviceId: string;
+
+  /**
+   * The Arn of the namespace that you want to use for DNS configuration.
+   */
+  readonly serviceArn: string;
+
+  /**
+   * The DnsRecordType used by the service
+   */
+  readonly dnsRecordType: DnsRecordType;
+
+  /**
+   * The Routing Policy used by the service
+   */
+  readonly routingPolicy: RoutingPolicy;
+
+  // Possibly add boolean loadbalancer field?
+}
+
+/**
+ * Basic props needed to create a service in a given namespace. Used by HttpNamespace.createService
+ */
+export interface BaseServiceProps {
+  /**
+   * A name for the Service.
+   *
+   * @default CloudFormation-generated name
+   */
+  name?: string;
+
+  /**
+   * A description of the service.
+   *
+   * @default none
+   */
+  description?: string;
+
+  /**
+   * Structure containing failure threshold for a custom health checker.
+   * Only one of healthCheckConfig or healthCheckCustomConfig can be specified.
+   * See: https://docs.aws.amazon.com/cloud-map/latest/api/API_HealthCheckCustomConfig.html
+   *
+   * @default none
+   */
+  healthCheckCustomConfig?: HealthCheckCustomConfig;
+}
+
+/**
+ * Service props needed to create a service in a given namespace. Used by createService() for PrivateDnsNamespace and
+ * PublicDnsNamespace
+ */
+export interface DnsServiceProps extends BaseServiceProps {
+  /**
+   * The DNS type of the record that you want AWS Cloud Map to create. Supported record types
+   * include A, AAAA, A and AAAA (A_AAAA), CNAME, and SRV.
+   *
+   * @default A
+   */
+  dnsRecordType?: DnsRecordType;
+
+  /**
+   * The amount of time, in seconds, that you want DNS resolvers to cache the settings for this
+   * record.
+   *
+   * @default 60
+   */
+  dnsTtlSec?: number;
+
+  /**
+   * A complex type that contains settings for an optional health check.
```

Comment:
I wouldn't mention "A complex type" here... This is about what it's for, not how it's modeled.